src/Symfony/Component/BrowserKit/Tests/ClientTest.php

```diff
@@ -94,13 +94,10 @@ public function testGetRequest()
         $this->assertEquals('http://example.com/', $client->getRequest()->getUri(), '->getCrawler() returns the Request of the last request');
     }
 
-    public function testGetRequestWithXHR()
+    public function testXmlHttpRequest()
     {
         $client = new TestClient();
-        $client->switchToXHR();
-        $client->request('GET', 'http://example.com/', array(), array(), array(), null, true, true);
-        $this->assertEquals($client->getRequest()->getServer()['HTTP_X_REQUESTED_WITH'], 'XMLHttpRequest');
-        $client->removeXHR();
+        $client->xmlHttpRequest('GET', 'http://example.com/', array(), array(), array(), null, true);
```

Comment:
in the end this test does not really test much. You can replace this with `$client->request(...)` and the test still passes.

Imho we should test that the client really receives a request with the correct header/server param?

Reply:
I agree, the test does not really test anything here.

Reply:
Doing it, I agree too.